packages/flutter_tools/lib/src/version.dart

```diff
@@ -752,24 +752,47 @@ class GitTagVersion {
         _runGit('git fetch $_flutterGit --tags', processUtils, workingDirectory);
       }
     }
-    // `--match` glob must match old version tag `v1.2.3` and new `1.2.3-dev.4.5`
-    return parse(_runGit('git describe --match *.*.* --first-parent --long --tags', processUtils, workingDirectory));
+    final List<String> tags = _runGit(
+      'git tag --contains HEAD', processUtils, workingDirectory).split('\n');
+    // Check first for a stable tag
+    for (final String tag in tags) {
+      final String trimmedTag = tag.trim();
+      if (RegExp(r'^\d+\.\d+\.\d+$').hasMatch(trimmedTag)) {
```

Comment:
I would hoist this and the other RegExp out of the loop. Then, document the format they are expected to match.

Is it possible to reuse some of the existing version parsing logic here?

Reply:
Good call on hoisting. I thought about reusing, but (at least in this file), they're all slightly different.

Reply:
Yeah, that is understandable - consider centralizing and standardizing them, so that we can easily verify that each version has the same level of documentation and testing.

Also would hopefully prevent anyone from accidentally re-implementing them

Reply:
well, I take that back, I can do a little reuse, lemme play with this...

Reply:
hoisted regexes. PTAL